### PR TITLE
Automatically build and test all possible combinations of packages

### PR DESCRIPTION
**Pull request by pbauer**
Do not merge yet. This is a proof of concept that it is possible to build and tests all possible combinations of packages. One problem is that running the tests will take quite some time (especially if we add more options) and travis with the limit of 10 minutes will not be enough.

### CODE REVIEW

File: test_all_combinations.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    return False
+
+
+def generate_all_combinations():
+    keys = SOURCE.keys()
+    first = SOURCE[keys[0]]
+    combinations = [[x] for x in first]
+    for key in keys[1:]:
+        combinations = combine(SOURCE[key], combinations)
+    combinations = [dict(zip(keys, x)) for x in combinations]
```
Comment (do3cc) [inline test_all_combinations.py]: I thought you don't like functional methos like zip and map and so on

Comment (pbauer) [inline test_all_combinations.py]: I like them when you write them for me.

**Comment by tisto** [comment]
@pbauer if you think that being able to run all those combinations is the main problem, you entirely missed the point I was trying to make in #46.

**Comment by tisto** [comment]
https://en.wikipedia.org/wiki/Unit_testing

**Comment by do3cc** [comment]
I don't know this guy

**Pull request closed without merging**